src/Core/Framework/Migration/Command/CreateMigrationCommand.php

```diff
@@ -21,21 +28,82 @@ class CreateMigrationCommand extends Command
     private $projectDir;
 
     /**
-     * @var PluginService
+     * @var KernelPluginCollection
      */
-    private $pluginService;
+    private $kernelPluginCollection;
 
     /**
-     * @var KernelPluginCollection
+     * @var EntityRepositoryInterface
      */
-    private $kernelPluginCollection;
+    private $pluginRepository;
 
-    public function __construct(KernelPluginCollection $kernelPluginCollection, PluginService $pluginService, string $projectDir)
-    {
+    public function __construct(
+        KernelPluginCollection $kernelPluginCollection,
+        string $projectDir,
+        EntityRepositoryInterface $pluginRepository
+    ) {
         parent::__construct();
         $this->projectDir = $projectDir;
-        $this->pluginService = $pluginService;
         $this->kernelPluginCollection = $kernelPluginCollection;
+        $this->pluginRepository = $pluginRepository;
+    }
+
+    public function completeOptionValues($optionName, CompletionContext $context)
+    {
+        if ($optionName === 'plugin') {
+            $criteria = new Criteria();
+            $criteria->addFilter(new EqualsFilter('managedByComposer', 0));
+
+            /** @var PluginCollection $pluginCollection */
+            $pluginCollection = $this->pluginRepository
+                ->search($criteria, Context::createDefaultContext())
+                ->getEntities()
+            ;
+            $result = [];
+
+            foreach ($pluginCollection as $plugin) {
+                if ($plugin->getPath() === null) {
+                    continue;
+                }
+
+                $pathNodes = explode(DIRECTORY_SEPARATOR, $plugin->getPath());
+                $result[] = array_pop($pathNodes);
+            }
+
+            return $result;
+        }
+
+        return [];
+    }
+
+    public function completeArgumentValues($argumentName, CompletionContext $context)
+    {
+        if ($argumentName === 'directory') {
+            $criteria = new Criteria();
+            $criteria->addFilter(new EqualsFilter('managedByComposer', 0));
+
+            if (($pluginNameIndex = array_search('--plugin', $context->getWords(), true)) !== false) {
+                $pluginName = $context->getWordAtIndex($pluginNameIndex + 1);
+                $criteria->addFilter(new ContainsFilter('path', $pluginName));
+            }
+
+            /** @var PluginCollection $pluginCollection */
+            $pluginCollection = $this->pluginRepository
+                ->search($criteria, Context::createDefaultContext())
+                ->getEntities()
+            ;
+            $result = [];
+
+            foreach ($pluginCollection as $plugin) {
+                $result[] = $plugin->getPath();
+            }
+
+            return $result;
+        } elseif ($argumentName === 'namespace') {
+            // TODO find good way to determine the namespace
```

Comment:
Hey @JoshuaBehrens,
will you still take care of that tone? :-)

Reply:
Good catch. I think by now it is easier for me to find namespaces :) I'll amend that